pkg/scheduler/eventhandlers.go

```diff
@@ -387,7 +403,7 @@ func AddAllEventHandlers(
 			FilterFunc: func(obj interface{}) bool {
 				switch t := obj.(type) {
 				case *v1.Pod:
-					return !assignedPod(t) && responsibleForPod(t, schedulerName) && !vmPodShouldSleep(t)
+					return !assignedPod(t) && responsibleForPod(t, schedulerName) && !vmPodShouldSleep(t) && podIsNetworkReady(t)
```

Comment:
Not sure if it's necessary to couple "networkReadiness" with scheduling. This might created some undesired situation: 1) Potential deadlocks, some network controllers may need scheduling decision in order to finish network resource creation. If a controller sets such work into two phases, it's complicated and will cause problem for kubelet  2) scheduler now has a dependency on external network controllers.

Reply:
This annotation is purposed for holding scheduler from premature schedulings. If no controller updates the annotation, yes the pod will not be scheduled. Such synchronization mechanism is collaborative between controller and scheduler; so each one has to fulfill its responsibility.

Regarding your points of concerns,
1) in that case, yes task should be split into pre-scheduling and other (concurrent-scheduling or post-scheduling); if kubelet has problem handling this, it will have problem handling w/o such mechanism;
2) yes scheduler has a dependency (coupling) on the collaborative controller. If this depency is undesired, we can consider using a switch (like scheduler startup param or feature gate) to turn it on/off.